### DISCUSSION TITLE
Unable to install Vineyard

### DISCUSSION THREAD

**Discussion by adwitiyaM1** [v6d-io/v6d#1020] [Q&A] [answered]
I'm trying to install Vineyard using the command: **pip install vineyard** , but I'm getting this error: 
ERROR: Could not find a version that satisfies the requirement vineyard (from versions: none)
ERROR: No matching distribution found for vineyard

I have Python 3.10.7 installed on Windows.

**Comment by sighingnow** [comment, marked as answer]
Hi Adwitiya @2022H1030042G,

Vineyard currently can only be installed on *nix platforms (Linux and MacOS). On Windows it could be installed inside [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) and native Windows is not supported yet.

**Comment by adwitiyaM1** [comment]
Okay, I'm using Virtual box now and installed Ubuntu. I have followed these steps: [https://v6d.io/notes/install.html](url) till "Install libgrape-lite". Now under "Building Vineyard" where do I have to follow these steps. Already build directory was created in the above steps. What does out-of-source build using CMake means?

**Reply by sighingnow** [reply]
> What does out-of-source build using CMake means?

It means that the generated artifacts, e.g., generated `.o`, `.so`, etc. are placed in the "build" directory.

**Comment by adwitiyaM1** [comment]
After completing the steps listed under "Install libgrape-lite" , now to follow the steps under "Building vineyard", under which path do I have to create a "build" folder.
I tried following these steps but I got "The source does not appear to contain CMakeLists.txt" error. Kindly help me to resolve it.

**Reply by dashanji** [reply]
Have you run `git submodule update --init` in the v6d root directory?
